src/system.c

```diff
@@ -25,6 +25,17 @@ void emu_update_uart_interrupts(riscv_t *rv)
     plic_update_interrupts(attr->plic);
 }
 
+/*
+ * Linux kernel might create signal frame when returning from trap
+ * handling, which modifies the SEPC CSR. Thus, the fault instruction
+ * cannot always redo. For example, invalid memory access causes SIGSEGV.
+ */
+extern bool need_handle_signal;
+#define CHECK_PENDING_SIGNAL()                                    \
```

Comment:
Specify `rv` parameter in this function-like macro. That is,
```c
#define CHECK_PENDING_SIGNAL(rv) \
    ...
```

Reply:
Instead of using `rv`, it may be more appropriate to use `need_handle_signal` as the parameter of the macro. This is because `rv` has only a single instance, and changes to its members are part of the macro's internal logic.

On the other hand, `need_handle_signal` is explicitly modified by the macro and directly determines whether the function shall return. This approach improves intuitiveness and readability. Consider the following revised version:
```c
#define CHECK_PENDING_SIGNAL(flag)        \
do {                                                                 \
    flag = (rv->csr_sepc != rv->last_csr_sepc); \
} while (0);
```